doc/mutations/creating-mutation-plugins.md

```diff
@@ -1,7 +1,192 @@
-## Creating mutation plugins
+# Creating mutations for Entities
+The graphql module uses the [Drupal plugin system](https://www.drupal.org/docs/8/api/plugin-api/plugin-api-overview) for a lot of the extensibility features of plugins. So a lot of the times when you want to extend the graphql module (for examply when creating your own mutations) you will be using the plugin system and creating your own plugins.
 
-WIP : For now for relevant information on custom mutations check the following links : 
+## Why the automatic mutations where removed. 
 
--  https://www.amazeelabs.com/en/blog/extending-graphql-part-3-mutations
--  https://github.com/justinlevi/graphql_custom_mutation
--  http://joaogarin.com/posts/drupal-graphql-with-angular-and-apollo-part3
+In [this article](https://www.amazeelabs.com/en/blog/extending-graphql-part-3-mutations) it's well explained why automatic mutations where removed. But this, as stated in the article, does not mean that creating mutations is complicated. In fact, its a simple task and one that might even provide with the extra flexibility you know and love from Drupal.
+
+## Mutations to create Drupal Entities
+One of the most common things you will want to do when with mutations is to create, update and delete entities. These CRUD operations on entities where made as simple as possible to implement with the module by extending some generic classes provided by the graphql_core module.
```

Comment:
"When with mutations"?

Reply:
"W*h*ere made as simple as possible"

Reply:
Just say that "were made as simple as possible to implement and only require you to extend some generic classes provided by the graphql_core module"